clients/js/README.md

```diff
@@ -1 +1,181 @@
-# Raccoon js client
+# Raccoon Client for Javascript
+
+A JS client library for [Raccoon](https://github.com/raystack/raccoon), compatible both in browsers as well as nodejs environment.
+
+Features:
+* Send JSON/Protobuf encoded messages to raccoon server
+* Configurable request wiretype i.e.,JSON/PROTOBUF
+* Retry mechanism
+
+**Note:** For encoding protobuf messages, this client utilises [protobufjs](https://github.com/protobufjs/protobuf.js).
+
+## Requirements
+
+- **Node.js**: Version 20.x or above.
+- **Browser**: Modern web browser with ES6+ compatibility.
+
+## Install
+
+```bash
+npm i @raystack/raccoon --save
+```
+
+## Usage
+
+#### Construct a new REST client, then pass the available options on the client.
+
+For example:
+
+```javascript
+import { RaccoonClient, SerializationType, WireType } from '@raystack/raccoon';
+```
+
+```javascript
+const raccoonClient = new RaccoonClient({
+    serializationType: SerializationType.PROTBUF,
+    wireType: WireType.JSON,
+    timeout: 5000,
+    url: 'http://localhost:8080/api/v1/events',
+    headers: {
+        'X-User-ID': 'user-1'
+    }
+});
+```
+
+#### Send messages
+
+In the below example, we are sending protobuf messages. 
+
+**Note:** In this case, we only need to send the protobufjs object along with values, the client will take care of encoding the object.
+
+```javascript
+const pageEvent = new clickevents.PageEvent();
+pageEvent.eventGuid = "123";
+pageEvent.eventName = "page open";
+
+const clickEvent = new clickevents.ClickEvent();
+clickEvent.eventGuid = "123";
+clickEvent.componentIndex = 1;
+clickEvent.componentName = "images";
+
+const events = [
+    {
+        type: 'test-topic1',
+        data: clickEvent
+    },
+    {
+        type: 'test-topic2',
+        data: pageEvent
+    }
+];
+
+raccoonClient.send(events)
+    .then(result => {
+        console.log('Result:', result);
+    })
+    .catch(error => {
+        console.error('Error:', error);
+    });
+```
+
+**Note:** Here, the PageEvent and ClickEvent are imported from generated protobufjs code.
+
+See complete examples for sending JSON and protobuf messages: [examples](examples)
+
+## Configurations
```

Comment:
Won't a tabular format be easier to read and follow. The line by line on readme, looks a bit sparse.

Reply:
I am following the usual config format that in being followed in raytstack

example: https://github.com/raystack/raccoon/blob/main/docs/docs/reference/configurations.md